### PR TITLE
Fix branch-level conflicts on `git export` (#463)

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)

### CODE REVIEW

File: tests/snapshots/test_git_colocated__e61b6729ff4292870702f2f72b2a60165679ef37.snap [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (yuja) [inline tests/snapshots/test_git_colocated__e61b6729ff4292870702f2f72b2a60165679ef37.snap:1]: Nit: unused insta snapshots?

File: lib/src/git.rs [hunk 2]
```diff
@@ -232,7 +245,7 @@ pub fn export_changes(
             git_ref.delete()?;
         }
     }
-    Ok(())
+    Ok(exported_view)
```
Comment (yuja) [inline lib/src/git.rs:254]: Here we build a separate `exported_view` in order to minimize the serialized data? After "git: update our record of Git branches on export", `mut_repo.view()` should also have a valid data.

Just wanna confirm if I understand things correctly.

**Review by yuja — APPROVED** [review]

File: tests/snapshots/test_git_colocated__e61b6729ff4292870702f2f72b2a60165679ef37.snap (hunk 1 above)
Comment (martinvonz) [reply tests/snapshots/test_git_colocated__e61b6729ff4292870702f2f72b2a60165679ef37.snap]: Oops, fixed.

File: lib/src/git.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/git.rs:254]: Yes, good point! I had not even thought of that. I think you're right that we should be able to remove this extra `exported_view` at the end of this series. I'll look into adding a commit for that.

Comment (martinvonz) [reply lib/src/git.rs:254]: Wait, no, `mut_repo.view()`'s branches will not be the same what we exported to Git if any of the branches had conflicts. When there are branches with conflicts, the `exported_view` will have those branches left at an older state. That way, we try to export that branch again on subsequent exports, until it no longer has a conflict.

However, `mut_repo.view()`'s Git refs should match the export state, so I think we should be able to compare the exported Git refs to current local branches. I'll have to think a bit more about that.

**Pull request merged**